src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,29 @@
+package seedu.addressbook.data.person;
+
+public class Contact {
+	
+	public String value;
+	protected boolean isPrivate; //Changed private to protected 
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Address // instanceof handles nulls
```

Comment:
hmm. this is the Contact class. Now we can't check if an `Email` object is equal to another `Email` object.